### ISSUE TITLE
Payments method throws server response instead of Error

### ISSUE THREAD

**Issue by 0x80** [mollie/mollie-api-node#101]
A call to `client.payments.create()` throws an object containing the server response. 

```
{ status: 422,
  title: 'Unprocessable Entity',
  detail: 'A customer is required for this recurring type',
  field: 'customerId',
  _links: 
   { documentation: 
      { href: 'https://docs.mollie.com/guides/handling-errors',
        type: 'text/html' } } }
```

I think throwing anything but an Error object is considered bad practice. Is this done on purpose or should this be considered a bug?

I am guessing this is not the only method with this behaviour. If it is intentional it would be helpful if it's documented.

**Comment by adamreisnz** [comment]
This is causing Node errors in Node v10:

`Warning: a promise was rejected with a non-error`

Proper errors should be thrown rather than response objects. The response object can be appended to the error as `error.response`.

**Comment by vernondegoede** [comment]
Thanks @adamreisnz and @0x80. I agree that we should refactor this and throw an error (containing this information) instead of returning an object.

However, I'd like to wait for our TypeScript rewrite to land in master before changing this.

**Comment by outreal** [comment]
[VError](https://github.com/joyent/node-verror) does not like it too.

As the response is not a proper Error, I have to encapsulate it into a new Error object, which is rather non-trivial.

**Comment by outreal** [comment]
Comment to keep it from closing... ><

**Comment by Pimm** [comment]
Hey @outreal, we're currently planning on merging a branch which will fix this. I'll keep you posted!

**Comment by vernondegoede** [comment]
We've just released a beta version of the Node API client 3.0.0, which comes with the following changes:

- Added TypeScript type definitions
- Added Orders, Shipments and Captures APIs
- Added ES module in addition to the CommonJS module
- Added QR code support
- Added pagination support
- Added support for embedded resources (`payments`, `refunds`, `chargebacks`, etc.)
- Improved exceptions

Check out the [changelog](https://github.com/mollie/mollie-api-node/blob/master/CHANGELOG.md)
Check out the [migration guide](https://github.com/mollie/mollie-api-node/blob/master/MIGRATION.md)

Feel free to test the beta and let us know if you have any feedback!

```
$ yarn add @mollie/api-client@beta
```
